set.mm

```diff
@@ -662596,6 +663406,1223 @@ closed under the multiplication ( ' X ' ) of a finite number of
+      mpd xrltso infex rgenw fnmpt breq1 ralrn supxrleub mp2an supxrss xrletri3
+      cvv sylibr sylanbrc eqtrd ) ADUAUBZEUCZNOUDZEBUEZNOUDZADFPXRXTUFHKDEFECQD
+      CUGZUHUIUJZUEZNUKZNOULZVAKQDKUGZUHUIUJZUEZNUKZNOULZVAGCKQYGYLYCYHUFZNYFYK
+      OYMYEYJNYMYDYIDYCYHUHUIUMUNUOUPUQURZUSUTAXTYBRSZYBXTRSZXTYBUFZALUGZYBRSZL
+      XSVBZYOAMUGZEUBZYBRSZMQVBZYTAUUAYRRSZLBVCZMQVBZUUDAUUGBNOUDUHUFZJABNVLUUG
+      UUHVDABIVEMLBVFUTVGAUUFUUCMQAUUAQPZTZUUEUUCLBUUJYRBPZUUETZTZUUBYREUBZYBUU
+      IUUBNPAUULQNUUAEKDEYNVHZVIVMUUMAUUKUUNNPZAUUIUULVJZUUJUUKUUEVKZAUUKTZYRQP
+      ZUUPABQYRIVNZQNYREUUOVIUTVOYBNPZUUMYANVLZUVBYAXSNEBVPZQNEVQXSNVLZUUOQNEVR
+      VSZVTZYAWAVSZWBUUMUUBUUNRSZDUUAUHUIUJUENUKNOULZDYRUHUIUJUENUKNOULZRSZUUMU
+      UIUUTUUEUVLAUUIUULWCUUMAUUKUUTUUQUURUVAVOUUJUUKUUEWDUUAYRDWEWFUUJUUKUVIUV
+      LVDUUEUUJUUKTUUBUVJUUNUVKRUUIUUBUVJUFAUUKKDEUUAYNWGVMAUUKUUNUVKUFZUUIUUSU
+      UTUVMUVAKDEYRYNWGUTWHWIWJVGUUMAUUKUUNYBRSZUUQUURUUSUVCUUNYAPUVNUVCUUSUVGW
+      BUUSQYRBEAEQWMZUUKAKQYLENAKWNYLNPZAYHQPTYKNVLUVPYJNWKYKWOVSWBYNWLWPUVAAUU
+      KWQWRYAUUNWSVOVOWTXAXBXCUVOYTUUDVDYLXNPZKQVBUVOUVQKQNYKOXDXEXFKQYLEXNYNXG
+      VSYSUUCLMQEYRUUBYBRXHXIVSXOUVEUVBYOYTVDUVFUVHLXSYBXJXKXOAYAXSVLZUVEYPUVRA
+      UVDWBUVEAUVFWBYAXSXLVOXTNPZUVBYQYOYPTVDUVEUVSUVFXSWAVSUVHXTYBXMXKXPXQ $.
+  $}
+
+  ${
+    climlimsupcex.1 $e |- -. M e. ZZ $.
+    climlimsupcex.2 $e |- Z = ( ZZ>= ` M ) $.
+    climlimsupcex.3 $e |- F = (/) $.
+    $( Counterexample for ~ climlimsup , showing that the first hypothesis is
+       needed, if the empty set is a complex number (see ~ 0ncn and its
+       comment) (Contributed by Glauco Siliprandi, 2-Jan-2022.) $)
+    climlimsupcex $p |- ( ( (/) e. CC /\ -. -oo e. CC ) -> ( F : Z --> RR /\ F
+       e. dom ~~> /\ -. F ~~> ( limsup ` F ) ) ) $=
+      ( c0 cc wcel cmnf wn wa cr wf cli clsp cfv wbr wceq adantr cdm f0 cuz uz0
+      cz ax-mp eqtri feq12i mpbir a1i wrel climrel 0cnv releldm syl2anc adantlr
+      syl5eqbr simpr fveq2i limsup0 breq12i biimpi climuni nelneq ad2antrr 3jca
+      adantll pm2.65da ) GHIZJHIKZLZCMANZAOUAIZAAPQZORZKVLVKVLGMGNMUBCGMAGFCBUC
+      QZGEBUEIKVPGSDBUDUFUGUHUIUJVIVMVJVIOUKZAGORVMVQVIULUJVIAGGOFUMUQAGOUNUOTV
+      KVOGGORZVIVOVRVJVIVRVOUMTUPVKVOLVRGJSZVOVRVSVKVOVRLVRGJORZVSVOVRURVOVTVRV
+      OVTAGVNJOFVNGPQJAGPFUSUTUGVAVBTGJGVCUOVGVKVSKVOVRGJHVDVEVHVHVF $.
+  $}
+
+  ${
+    liminfcld.1 $e |- ( ph -> F e. V ) $.
+    $( Closure of the inferior limit.  (Contributed by Glauco Siliprandi,
+       2-Jan-2022.) $)
+    liminfcld $p |- ( ph -> ( liminf ` F ) e. RR* ) $=
+      ( wcel clsi cfv cxr liminfcl syl ) ABCEBFGHEDBCIJ $.
+  $}
+
+  ${
+    $d F k $.  $d Z k y $.  $d k ph y $.
+    liminfresico.1 $e |- ( ph -> M e. RR ) $.
+    liminfresico.2 $e |- Z = ( M [,) +oo ) $.
+    liminfresico.3 $e |- ( ph -> F e. V ) $.
+    $( The inferior limit doesn't change when a function is restricted to the
+       upper part of the reals.  (Contributed by Glauco Siliprandi,
```

Comment:
Same comment as above.

Reply:
New commit, now it is:
The inferior limit doesn't change when a function is restricted to an upperset of reals.